### ISSUE TITLE
nodejs connector?

### ISSUE THREAD

**Issue by jlaustill** [simogeo/Filemanager#507]
Is anyone working on a nodejs connector?  I'm going to need one for an upcoming migration from mvc.  If nobody is working on one that I can help with, I will be writing one :)

**Comment by servocoder** [comment]
Thank you for your interest!
It would be great to have nodejs connector. You can become a maintainer of it.
But this repo is weakly supported currently. I can suggest you to migrate to [Rich Filemanager](https://github.com/servocoder/RichFilemanager). All new features are implementing there. You can create new issue there, and use it for any further issues and suggestions, please.

**Comment by jlaustill** [comment]
I had NO idea this was deprecated, I see the note in the readme now that you mention.  Would it be possible to spell that out a bit better?  I use this project HEAVILY at work, so I'll check out the new fork today and go from there.  Thanks.

**Comment by servocoder** [comment]
Ok, I will emphsize it in more explicit form.
You are welcome. Feel free to create new issue when you are going to start implement a nodejs connector, so others could see that work is in process.

**Comment by simogeo** [comment]
Indeed, it is now time to switch to  RichFilemanager ! @servocoder did a great work and will maintain the package.
I added a clear notice on the project description. See : https://github.com/simogeo/Filemanager

@jlaustill : will be great to have a nodejs package maintainer !

**Comment by servocoder** [comment]
Thanks @simogeo, you did a great job for the FM.
All I do is based on the things which you had implemented initially.

**Comment by xcapt** [comment]
Hi,

just open sourced express-filemanager which is a nodejs / express connector for simogeo's filemanager. 

Happy to welcome contributor to migrate it to Rich Filemanager !!

**Comment by servocoder** [comment]
Hi @xcapt 
Rich Filemanager has a maintainer for nodejs - @jlaustill. Check out this thread: https://github.com/servocoder/RichFilemanager/issues/30

I don't know if @jlaustill still works on the nodejs connector for the RFM, he haven't replied on my last comment. He brought a lot of great ideas on how to improve the RFM, so I hope he will come back.

**Comment by jlaustill** [comment]
Hey @servocoder, Sorry, I didn't see your last comment.  My use case has moved so far from the original filemanager I'm not sure if I will be able to contribute back.  I'm pretty sure not even 1 line of code matches in my repo anymore to get this working with my use case and to pass my code reviews.  

Making this work client/api has been way more complicated than I originally expected.  So this ended up being basically a complete rewrite for me.  I just deployed it last week and it seems to be working well for people, but it's nowhere NEAR done yet.

You can check out what I've done here [FM2](https://github.com/jlaustill/Filemanager-2.0) .

**Comment by servocoder** [comment]
Hi @jlaustill 
Great work on FM2
You are right, there were a lot of changes indeed. And some are coming soon. You have seen the roadmap and I constantly working on Rich Filemanager to make it better. Currently I'm working on JSON API standardization as you suggested and the implementation of new filetree plugin for the further features. Also there are other great features in todo list like clipboard support and multiple files/folders actions. If you are going to continue with your own branch exclusively you can lost all those features. Anyhow let me know whether you are going to maintain nodejs connector  for Rich Filemanager in the future or not. Thank you.

**Comment by jlaustill** [comment]
Hey @servocoder ,

I just looked at @xcapt 's code, and I think it would be a much better fit for rich filemanager.  He kept his code almost 100% in line with the PHP connector.  I wonder if he'd be willing to move forward with it?

I am so busy right now I am struggling to just keep up, but once my FM2 is stable I could certainly work on updating the nodejs connector for rich filemanager if @xcapt can't.  

FM2 meets a completely different need than the original filemanager.  It's more for an intranet or internal web app, not so much for public facing CMS's.  I went forward with it super fast because I needed something to deploy.  The more I work on it though, the more I'm finding that the two use cases are pretty different.  I have no need for clipboard support, drag and drop, etc etc.  I just need the basic basics.  But I do need CORS support, role based secure folders, and the ability to upload 100+ files at once.  These are the things that work in FM2 already.  

I would love to figure out how to bring these two use cases together in future, but I'm still learning all the ins and outs, and they still just seem very very different.

**Comment by servocoder** [comment]
@jlaustill I see, either way I will keep you in touch, especially when the structural changes are complete, so perheps you will be able to help with nodejs connector. Gool luck :)

**Comment by jlaustill** [comment]
@servocoder sounds great!
